docs/BasicCommunication/OnPutFile/index.md

```diff
@@ -13,27 +13,32 @@ Purpose
 
 !!! MUST
 
-Use the appropriate uploaded file according to its workflow (IVSU, SystemRequest, RPC's). See diagrams listed below.
+1. Use the appropriate uploaded file according to its workflow (IVSU, SystemRequest, RPC's). See diagrams listed below.  
+2. Whenever HMI gets RPC with `Image` which has `.isTemplate` set to true, the HMI has to:
 
+- load the proper image pattern
+- extract alpha channel from the template image
+- apply the alpha channel to the image pattern
+- use this newly generated image instead of the uploaded one  
+
+and whenever the UI changes, the HMI has to recreate images currently visible on the screen.
 !!!
 
 !!! NOTE
 
 The list of RPCs and data structures that `OnPutFile` affects are:
 
   * Show (Image, SoftButton)
-  * ShowConstantTBT (SoftButton)
+  * ShowConstantTBT (Image, SoftButton)
   * CreateInteractionChoiceSet (Image)
-  * DeleteInteractionChoiceSet (Image)
   * SetGlobalProperties (Image, VrHelpItem)
   * ResetGlobalProperties (Image, VrHelpItem)
-  * UpdateTurnList (Turn)
+  * UpdateTurnList (Turn, SoftButton)
   * AddCommand(Image)
```

Comment:
AddSubMenu can also have an image

Reply:
@yang1070, AddSubMenu got `menuIcon` param not in terms of **Template images** proposal, 
but in terms of further **SubMunu_icon** feature. 
I guess, it would be incorrect to add AddSubMenu on the list in terms of this PR  
and I am not sure that the code covers this.